### PR TITLE
SKY-13 

### PR DESCRIPTION
**Pull request by adamveszelinov**
Please review it. Thanks.

### CODE REVIEW

File: skyxplore-ui-test/src/test/java/com/mentoring/pageobject/CharacterSelectionPage.java [hunk 1]
```diff
@@ -141,6 +141,20 @@ public CharacterSelectionPage clickRenameCharacter(String characterName) {
...
     }
 
+    @Step("Rename from: {characterNameOld} - and choose new character: {characterNameNew}.")
+    public OverviewPage renameAndChooseCharacter(String characterNameOld, String characterNameNew) {
+        clickRenameCharacter(characterNameOld);
+        renameCharacter(characterNameNew);
+        return selectCharacter(characterNameNew);
+    }
+
+    @Step("")
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/pageobject/CharacterSelectionPage.java]: Please fill the step description or just remove: clickForDeleteCharacterButtonInList is already annotated with Step. If you decide to have them both annotated, then please re-think the descriptions (one is about just clicking, the other is actually performing the deletion)

File: skyxplore-ui-test/src/test/java/com/mentoring/pageobject/CommunityPage.java [hunk 2]
```diff
@@ -95,8 +103,31 @@ public CommunityPage clickFriendRequestsButton() {
...
+
+    @Step("Asserting if character name is marked as invalid.")
+    public boolean isFriendRequestListEmpty() {
+        return isElementDisplayed(noFriendRequest);
+    }
+
+    @Step("Clicking on the received friend request refuse button: {characterName}.")
+    public CommunityPage clickReceivedFriendRequestDeclineButton(String characterName) {
+        Actions action = new Actions(driver);
+        WebElement we = waitUntilVisible(xpath(String.format(APPROVE_RECEIVED_FRIEND_REQUEST_BY_NAME_SELECTOR, characterName)));
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/pageobject/CommunityPage.java]: Please use meaningful names (webElement, element)

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 3]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
+                .renameAndChooseCharacter(characterNameB, characterNameBMod)
+                .openCommunityPage()
+                .clickFriendRequestsButton()
+                .clickReceivedFriendRequestApproveButton(characterNameA)
+                .logout()
+                .login(userA)
+                .selectCharacter(characterNameA)
+                .openCommunityPage();
+
+        softAssertion.assertThat(communityPage.isCharacterInFriendList(characterNameBMod)).isTrue();
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Please add assertion message (.as())

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 4]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
+
+    @Description("Negative test case 2: " +
+            "User B has a pending request from user A. " +
+            "User B renames the character he got the request with, " +
+            "then accepts the request.")
+    @Test(groups = SMOKE)
+    @Severity(SeverityLevel.BLOCKER)
+    @Feature(FRIEND_REQUEST)
+    public void renameCharacterAndAcceptFriendRequest() {
+
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Please remove unnecessary empty line

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 5]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
+        softAssertion.assertAll();
+    }
+
+    @Description("Negative test case 2: " +
+            "User B has a pending request from user A. " +
+            "User B renames the character he got the request with, " +
+            "then accepts the request.")
+    @Test(groups = SMOKE)
+    @Severity(SeverityLevel.BLOCKER)
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Blocker level is an exaggeration in my opinion, as it's not a general use case (rather edge-case). Please revisit the severity levels in this class (blocker=major functionality is not working at all, critical path cannot be completed successfully)

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 6]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
+
+    @Description("Negative test case 3: " +
+            "User B has a pending request from user A. " +
+            "Delete user B's character " +
+            "and check the request at user A.")
+    @Test(groups = SMOKE)
+    @Severity(SeverityLevel.BLOCKER)
+    @Feature(FRIEND_REQUEST)
+    public void sendFriendRequestAndDeleteRequestedCharacter() {
+
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Please remove unnecessary empty line

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 7]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
+    @Description("Negative test case 4: " +
+            "User B has a pending request from user A. " +
+            "Delete user A's character " +
+            "and check the request at user B.")
+    @Test(groups = SMOKE)
+    @Severity(SeverityLevel.BLOCKER)
+    @Feature(FRIEND_REQUEST)
+    public void sendFriendRequestAndDeleteSenderCharacter() {
+
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Please remove unnecessary empty line

File: skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java [hunk 8]
```diff
@@ -102,4 +109,109 @@ public void sendAndReceiveFriendRequest() {
...
 
+    @Description("Negative test case 1: " +
+            "User A and user B exist. " +
+            "User A has already sent friend request to user B's character on the backend. " +
+            "User B logs in and refuses user A's request.")
+    @Test(groups = SMOKE)
+    @Severity(SeverityLevel.BLOCKER)
+    @Feature(FRIEND_REQUEST)
+    public void sendAndDeclineFriendRequest() {
+
```
Comment (gergelyszanto) [inline skyxplore-ui-test/src/test/java/com/mentoring/test/FriendRequestTest.java]: Please remove unnecessary empty line

**Review by gergelyszanto — DISMISSED** [review]

**Review by gergely-szanto-epam — APPROVED** [review]
